@shared/api/helpers.ts

```diff
@@ -4,17 +4,24 @@ import {
   EXTERNAL_MSG_RESPONSE,
 } from "../constants/services";
 import { Response } from "./types";
-import { TimeoutError } from "../constants/errors";
+import { NoExtensionInstalledError } from "../constants/errors";
+
+declare global {
+  interface Window {
+    lyra: boolean;
+  }
+}
 
 export const sendMessageToContentScript = (msg: {}): Promise<Response> => {
   window.postMessage(
     { source: EXTERNAL_MSG_REQUEST, ...msg },
     window.location.origin,
   );
   return new Promise((resolve, reject) => {
-    const timeout = setTimeout(() => {
-      reject(new TimeoutError());
-    }, 1000);
+    if (!window.lyra) {
+      reject(new NoExtensionInstalledError());
```

Comment:
if we want to short circuit based on Lyra not being installed, we can simply check to see if Lyra's content script has been installed